py/bc.h

```diff
@@ -84,6 +84,10 @@ typedef struct _mp_code_state_t {
     #if MICROPY_STACKLESS
     struct _mp_code_state_t *prev;
     #endif
+    #if MICROPY_PY_SYS_TRACE
+    struct _mp_code_state_t *prev_state;
```

Comment:
There's already `struct _mp_code_state_t *prev;` above.

Reply:
The **prev_state** is *MICROPY_STACKLESS* agnostic which **prev** is not. I tried to use the **prev** but it was difficult to change the logic around it to separate the normal execution traceback from settrace callback execution which is needed for the **frame** object. In this sense they are not the same. Also this feature is not mature so it seems easier to keep it's members separate from the uPy core (easier to debug).